x-pack/plugins/serverless/public/plugin.tsx

```diff
@@ -64,7 +63,8 @@ export class ServerlessPlugin
 
     return {
       setSideNavComponent: (sideNavigationComponent) =>
```

Comment:
nit: I've always disliked abbreviations, (and `Component` feels redundant, as the parameter is a `React` component)... perhaps we could consider `setSidebarNavigation`...?

Reply:
I agree w/ you, but `serverless.setSideNavComponent` already has presence in the 3 serverless project plugins. Let's find a new name at our next sync and change this in a standalone PR.

Reply:
`setSidebarNavigation()` is reserved (™) and will be used when there is no need to provide a component (UI) (define the navigation through an object - see first example in Proposal https://github.com/elastic/kibana/issues/157702).
Well actually it will even be shorter: `setNavigation()` 😊 

I do think "Component" is explicit of the intent. It could be "UI" suffix instead.